docs/proposals/resource-deletion-protection/README.md

```diff
@@ -0,0 +1,162 @@
+
+User can add a label `karmada.io/deletion-protected=true` to the resource that needs protection. When the resource with this label is being deleted, we will interrupt the user's delete request.
+
+#### Design Details
+
+##### Add the protected label
+`kubectl label <resource-type> <resource-name> karmada.io/deletion-protected=true`
+
+`karmadactl protect <resource-type> <resource-name>`
+
+##### Delete the protected label
+`kubectl label <resource-type> <resource-name> karmada.io/deletion-protected=false`
+
+`karmadactl unprotect <resource-type> <resource-name>`
+
+##### Prevent deletion operation
+
+For preventing deletion operations, we can achieve it through an Admission webhook. When a user attempts to delete, we need to first check if the label has a value of "true". If it does, we will interrupt the user's deletion operation.
+
+#### Alternatives
+
+##### Why choose Label over Annotations to tag resources?
+
+Annotations are typically used to store metadata that is not used for identification and selection, such as descriptive information or the status information of management tools. In this case, labels like "cannot be deleted" or "protected status" are not suitable as annotations.
+
+##### What will happen If user want to delete resource by `--force`?
+
+We may not be able to distinguish between the force delete and normal delete operations, so for force delete, we will still follow the original logic and prevent users from deleting the resource.
+
+##### Enable protection by default
+
+We can enable deletion protection by default for all resources, if user want to delete the resources, he must remove the label first.
+
+#### Pros
+
+- Easy to implement
+- Lower resource consumption, no need to watch certain resources.
+
+#### Cons
+
+- The user needs to add/remove tags to resources one by one.
+- The permission management is scattered, unable to follow the resources for permission control and unified handling.
+- A label is not a runtime state, and in certain situations, labels might be removed (such as when a controller is deleted and then the resource is re-created).
+
+### Proposal-2(add a CRD to specify the resources which should be protected)
+
+We can add a CRD `ResourceDeletionProtection` to specify the resources which should be protected.
+
+#### Design Details
+
+##### CRD example
+
+It's a tentatively designated as a cluster-level resource.
+
+```yaml
+kind: ResourceDeletionProtection
+metadata:
+  name: sample
+spec:
+  protectedResources:
+    - resourceType: Deployment
+      resourceNames:
+        - deployment-1
+        - deployment-2
+    - resourceType: Service
+      resourceNames:
+        - service-1
+```
+
+##### Webhook implement detail
+
+During startup, the webhook reads all 'ResourceDeletionProtection' entries into memory and consolidates them into a map. It then watches all changes to this resource, continuously updating the map. When a user deletes a resource, this map is checked to implement caching.
```

Comment:
Is it appropriate to add this kind of processing logic in a webhook? It seems like something that should be done by a controller. I think we could directly use an informer, like this: https://github.com/karmada-io/karmada/blob/7c96e0db541989bd95772bbc1cc0109d8479371c/pkg/webhook/resourceinterpretercustomization/validating.go#L34